sw/airborne/firmwares/rotorcraft/guidance/guidance_indi_hybrid_quadplane.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2015 Ewoud Smeur <ewoud.smeur@gmail.com>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+/**
+ * @file firmwares/rotorcraft/guidance/guidance_indi_hybrid_quadplane.c
+ *
+ */
+
+#include "firmwares/rotorcraft/guidance/guidance_indi_hybrid.h"
+#include "stabilization/stabilization_attitude_ref_quat_int.h"
+#include "filters/low_pass_filter.h"
+#include "state.h"
+#include "generated/modules.h"
+
+
+#ifndef GUIDANCE_INDI_THRUST_Z_EFF
+#error "You need to define GUIDANCE_INDI_THRUST_Z_EFF"
+#else
+float guidance_indi_thrust_z_eff = GUIDANCE_INDI_THRUST_Z_EFF;
+#endif
+
+#ifndef GUIDANCE_INDI_THRUST_X_EFF
+#error "You need to define GUIDANCE_INDI_THRUST_X_EFF"
+#else
+float guidance_indi_thrust_x_eff = GUIDANCE_INDI_THRUST_X_EFF;
+#endif
+
+
+float bodyz_filter_cutoff = 0.2;
+
+Butterworth2LowPass accel_bodyz_filt;
+
+
+
+/**
+ *
+ * Call upon entering indi guidance
+ */
+void guidance_indi_quadplane_init(void) {
+  float tau_bodyz = 1.0/(2.0*M_PI*bodyz_filter_cutoff);
+  float sample_time = 1.0 / PERIODIC_FREQUENCY;
+  init_butterworth_2_low_pass(&accel_bodyz_filt, tau_bodyz, sample_time, -9.81);
+}
+
+
+/**
+ * Low pass the accelerometer measurements to remove noise from vibrations.
+ * The roll and pitch also need to be filtered to synchronize them with the
+ * acceleration
+ * Called as a periodic function with PERIODIC_FREQ
+ */
+void guidance_indi_quadplane_propagate_filters(void) {
+   // Propagate filter for thrust/lift estimate
+  float accelz = ACCEL_FLOAT_OF_BFP(stateGetAccelBody_i()->z);
+  update_butterworth_2_low_pass(&accel_bodyz_filt, accelz);
+}
+
+
+
+
+/**
+ * Perform WLS
+ *
+ * @param Gmat Dynamics matrix
+ * @param a_diff acceleration errors in earth frame
+ * @param body_v 3D vector to write the control objective v
+ */
+void WEAK guidance_indi_calcg_wing(float Gmat[GUIDANCE_INDI_HYBRID_V][GUIDANCE_INDI_HYBRID_U], struct FloatVect3 a_diff, float body_v[GUIDANCE_INDI_HYBRID_V]) {
```

Comment:
A weak function only makes sense if you have a default implementation, but since you have made explicit modules for each vehicle types, it is not needed I think. Only one of them will implement this function.
Maybe, these modules can even be declared in conflicts in the module XML files.

Reply:
I did try to declare conflicts. The tail sitter implementation should indeed not be weak anymore. I will remove it. The settings for the quadplane on the other hand still need to be WEAK since the skewing quadplane overrides this in ctrl_eff modules.